src/sas/sascalc/fit/expression.py

```diff
@@ -212,7 +212,7 @@ def eval_expressions():
     # CRUFT: python < 3.0;  doc builder isn't allowing the following exec
     # https://stackoverflow.com/questions/4484872/why-doesnt-exec-work-in-a-function-with-a-subfunction/41368813#comment73790496_41368813
     #exec(functiondef, global_context, local_context)
-    eval(compile(functiondef, '<string>', 'exec'), global_context, local_context)
+    eval(compile(functiondef, functiondef, 'exec'), global_context, local_context)
```

Comment:
`<string>` is the default argument for when the code is not taken from a file, as the docs describe
https://docs.python.org/3/library/functions.html#compile
Is there a reason to use `functiondef` here?

Reply:
Since the code of the function is not in a file, setting `filename` to `functiondef` allows to get the code of the function via the `f_code.co_filename` attribute of the traceback object, which is then analzed to retrieve the faulty constraint.